contracts/sweep/SweepERC20.sol

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import "./TokenSweep.sol";
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+
+/**
+ * @title Adds the ability to sweep a token to a beneficiary address
+ *
+ * @dev
+ */
+abstract contract SweepERC20 is TokenSweep {
+    /**
+     * @notice Sweep the erc20 tokens to the beneficiary address
+     *
+     * @dev Needs access control implemented in the inheriting contract
+     **/
+    function sweepERC20Tokens(address token, uint256 amount) external virtual {
```

Comment:
Feel like this should also be `internal`?

Reply:
Interesting thought there 🤔 

Do we want anyone to be able to call the sweep function, or have it access controlled?

I suspect the answer is access controlled, as the listed beneficiary is unlikely to be setup to do anything based off a token transfer, meaning some form of external orchestration is needed.

Reply:
I think you're right about this needing to be `internal` and without `virtual`.

With how `override` works, you're not ensured that parent behaviour is actually invoked (as overriding permits changing the behaviour).

The implementation would then be a `external` function with access control and any guarding logic for not sweeping up tokens are expected to be in the contract.